imnodes.cpp

```diff
@@ -1861,35 +1808,55 @@ static void MiniMapUpdate()
     GImNodes->CanvasDrawList->PushClipRect(
         mini_map_rect.Min, mini_map_rect.Max, true /* intersect with editor clip-rect */);
 
-    // Get zoom scaling (0, 1]
-    const float scaling = GImNodes->MiniMapZoom;
-
     // Draw links first so they appear under nodes, and we can use the same draw channel
     for (int link_idx = 0; link_idx < editor.Links.Pool.size(); ++link_idx)
     {
         if (editor.Links.InUse[link_idx])
         {
-            MiniMapDrawLink(editor, link_idx, editor_center, mini_map_center, scaling);
+            MiniMapDrawLink(editor, link_idx);
         }
     }
 
     for (int node_idx = 0; node_idx < editor.Nodes.Pool.size(); ++node_idx)
     {
         if (editor.Nodes.InUse[node_idx])
         {
-            MiniMapDrawNode(editor, node_idx, editor_center, mini_map_center, scaling);
+            MiniMapDrawNode(editor, node_idx);
         }
     }
 
+    // Draw editor canvas rect inside mini-map
+    {
+        const ImU32 canvas_color = GImNodes->Style.Colors[ImNodesCol_MiniMapCanvas];
+        const ImU32 outline_color = GImNodes->Style.Colors[ImNodesCol_MiniMapCanvasOutline];
+        const ImRect rect = ScreenSpaceToMiniMapSpace(editor, GImNodes->CanvasRectScreenSpace);
+
+        GImNodes->CanvasDrawList->AddRectFilled(rect.Min, rect.Max, canvas_color);
+        GImNodes->CanvasDrawList->AddRect(rect.Min, rect.Max, outline_color);
+    }
+
     // Have to pop mini-map clip rect
     GImNodes->CanvasDrawList->PopClipRect();
 
-    // Reset callback info after use
-    GImNodes->MiniMapNodeHoveringCallback = NULL;
-    GImNodes->MiniMapNodeHoveringCallbackUserData = NULL;
+    bool mini_map_is_hovered = ImGui::IsWindowHovered();
+
+    ImGui::EndChild();
+
+    bool center_on_click =
+        mini_map_is_hovered &&
+        ImGui::IsMouseDown(ImGuiMouseButton_Left) &&
+        editor.ClickInteraction.Type == ImNodesClickInteractionType_None &&
+        !editor.Nodes.Pool.empty();
```

Comment:
One last remark, let's do this instead here.

```suggestion
        !GImNodes->NodeIdxSubmissionOrder.empty();
```

Why? `Nodes.Pool` can contain elements even when no nodes are on canvas. This happens when a user adds nodes and then removes them.

`NodeIdxSubmissionOrder` gets `push_back`ed with every call to `BeginNode` and gets cleared in `BeginNodeEditor`, so it's a better counter for how many nodes are on screen in this frame.

This is pretty contrived. In the future, we can get rid of those object pools entirely and use just `ImVector` for everything. I'm working on that currently, but it's a big change 😅

Reply:
Done. :)